sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/ShuffleExchangeExec.scala

```diff
@@ -83,9 +83,8 @@ case class ShuffleExchangeExec(
     new ShuffledRowRDD(shuffleDependency, readMetrics, partitionStartIndices)
   }
 
-  def createLocalShuffleRDD(
-      advisoryParallelism: Option[Int] = None): LocalShuffledRowRDD = {
-    new LocalShuffledRowRDD(shuffleDependency, readMetrics, advisoryParallelism)
+  def createLocalShuffleRDD(partitionStartIndices: Array[Array[Int]]): LocalShuffledRowRDD = {
```

Comment:
We need a better argument name. This is sharing the same argument name as createShuffledRDD, but the semantics are not completely same. It looks hard to understand the values in Array[Array[Int]].

Reply:
how about `partitionStartIndicesPerMapper`?

Reply:
partitionStartIndicesPerMapperArray ?

Reply:
I don't think we need "arrayOf"... otherwise we'd have to prefix/suffix all our parameter names with they types...

Reply:
As long as we have a clear javadoc, I don't even think change the name is necessary here.